src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenByAuthorizationCodeParameterBuilder.cs

```diff
@@ -85,6 +85,7 @@ internal override Task<AuthenticationResult> ExecuteInternalAsync(CancellationTo
         /// <param name="withSendX5C"><c>true</c> if the x5c should be sent. Otherwise <c>false</c>.
         /// The default is <c>false</c></param>
         /// <returns>The builder to chain the .With methods</returns>
+        [System.Obsolete("Set SendX5C once for this application using WithSendX5C on the ConfidentialClientApplicationBuilder object.", false)]
```

Comment:
I don't think we should obsolete the one on the AT calls. Both should work. We can change documentation to reflect the one on the cca object, but let's not break people.

Reply:
I agree with Jenny that this will be an additional method and need not obsolete

Reply:
Right, good for overrides if such a thing is needed.  I was expecting that this would be a warning. not an error, so not break people.  If the library setup to treat this as an error despite passing false into the Obsolete?

Reply:
This will emit a warning, not an error. We do this from time to time. It will not break people, but it will annoy them. 

This will warn ALL developers using `sendx5c` (1st parties only I guess) to upgrade. Some projects (e.g. Kusto) are very strict and have a 0 warnings policy. Others won't care. 

If we think it's wrong to set sendX5C at request level, let's deprecate it. But we must make sure to update our 1st party docs and that the obsolete message is very clear. 

@henrik-me - thoughts?

Aside: When we release MSAL 5 (we keep saying this for the past 3 years), we will delete all the deprecated APIs.

Reply:
+1 to Jenny: Let's not remove per request override and thus I see no reason to introduce a warning here. If anything an analyzer needs to be written to ensure recommended practices are followed.

(btw Java sends the X5C by default but allows to disable sending it)